src/Range/AbstractRange.php

```diff
@@ -132,6 +132,10 @@ public function containsRange(RangeInterface $range)
      */
     public function split($networkPrefix)
     {
+        if ($this instanceof Single){
```

Comment:
No need for clone (they are immutable).
Also, we should return a Subnet instance (or chance the phpdoc)

Reply:
i updated `phpdoc` on `RangeInterface`